src/river/river.py

```
@@ -67,240 +71,308 @@ def transform_bbox(x_min, x_max, y_min, y_max, from_epsg, to_epsg):
+                continue
+
+            sx, sy, start_z = feat["start_x"], feat["start_y"], feat["start_z"]
+            ex, ey, end_z = feat["end_x"], feat["end_y"], feat["end_z"]
+
+            if (
+                    sx is not None
+                    and sy is not None
+                    and start_z is not None
+                    and start_z == max_z
+            ):
+                if (sx, sy) not in end_points:
```

Ruff (0.11.9)

Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)